### PR TITLE
feat: refactor Nodes in Network to use ts and hooks

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

[Issue#324](https://github.com/ripple/explorer/issues/324)

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

### CODE REVIEW

**Comment by pdp2121** [comment]
Please just look at the Nodes.tsx file change for now while waiting for the bar chart PRs to merge.

File: src/containers/Network/Nodes.tsx
Comment (ckniffen) [inline src/containers/Network/Nodes.tsx:19]: You can leverage `useQuery` and `refetchInterval` to simplify this.

File: src/containers/Network/Nodes.tsx
Comment (ckniffen) [inline src/containers/Network/Nodes.tsx:41]: nodes is not a property of Map.

File: src/containers/Network/Nodes.tsx
Comment (ckniffen) [inline src/containers/Network/Nodes.tsx:40]: This is caused by the `connect` function in Map jacking up its type.  It claims there are no props on the component

File: src/containers/Network/Nodes.tsx
Comment (ckniffen) [inline src/containers/Network/Nodes.tsx:19]: You can have nodes come from this and return it in fetchData.  That will remove an extra useState and setNodes.

File: src/containers/Network/Nodes.tsx
Comment (pdp2121) [reply src/containers/Network/Nodes.tsx:40]: Can we fix this by modifying the connect function in Map?

Comment (ckniffen) [reply src/containers/Network/Nodes.tsx:40]: I tried doing that really quick.  It needs to be made typescript and there were a lot of errors when i did that.  Also we are using such and old version of redux i dont know if they type can be corrected.

Comment (pdp2121) [reply src/containers/Network/Nodes.tsx:40]: Hmm I guess it can be a separate refactor ticket then

**Comment by JST5000** [comment]
> Please just look at the Nodes.tsx file change for now while waiting for the bar chart PRs to merge.

For next time, if you set the PR to merge with the bar chart PR's branch instead of main, it'll give you the same nice diff, and as soon as the other branch is merged into main it will automatically also point at main :) 

Can be quite convenient for situations like this!

**Review by jonathanlei — APPROVED** [review]
LGTM

**Review by JST5000 — APPROVED** [review]
LGTM

**Review by ckniffen — APPROVED** [review]

**Pull request merged**
